// @base-org/account/dist/core/telemetry/events/scw-sub-account.js
import { store } from '../../../store/store.js';
import { ActionType, AnalyticsEventImportance, ComponentType, logEvent } from '../logEvent.js';
export const logSubAccountRequestStarted = ({ method, correlationId, }) => {
    logEvent('scw_sub_account.request.started', {
        action: ActionType.unknown,
        componentType: ComponentType.unknown,
        method,
        correlationId,
        enableAutoSubAccounts: store.subAccountsConfig.get()?.enableAutoSubAccounts,
    }, AnalyticsEventImportance.high);
};
export const logSubAccountRequestCompleted = ({ method, correlationId, }) => {
    logEvent('scw_sub_account.request.completed', {
        action: ActionType.unknown,
        componentType: ComponentType.unknown,
        method,
        correlationId,
        enableAutoSubAccounts: store.subAccountsConfig.get()?.enableAutoSubAccounts,
    }, AnalyticsEventImportance.high);
};
export const logSubAccountRequestError = ({ method, correlationId, errorMessage, }) => {
    logEvent('scw_sub_account.request.error', {
        action: ActionType.error,
        componentType: ComponentType.unknown,
        method,
        correlationId,
        errorMessage,
        enableAutoSubAccounts: store.subAccountsConfig.get()?.enableAutoSubAccounts,
    }, AnalyticsEventImportance.high);
};
export const logAddOwnerStarted = ({ method, correlationId, }) => {
    logEvent('scw_sub_account.add_owner.started', {
        action: ActionType.unknown,
        componentType: ComponentType.unknown,
        method,
        correlationId,
        enableAutoSubAccounts: store.subAccountsConfig.get()?.enableAutoSubAccounts,
    }, AnalyticsEventImportance.high);
};
export const logAddOwnerCompleted = ({ method, correlationId, }) => {
    logEvent('scw_sub_account.add_owner.completed', {
        action: ActionType.unknown,
        componentType: ComponentType.unknown,
        method,
        correlationId,
        enableAutoSubAccounts: store.subAccountsConfig.get()?.enableAutoSubAccounts,
    }, AnalyticsEventImportance.high);
};
export const logAddOwnerError = ({ method, correlationId, errorMessage, }) => {
    logEvent('scw_sub_account.add_owner.error', {
        action: ActionType.error,
        componentType: ComponentType.unknown,
        method,
        correlationId,
        errorMessage,
        enableAutoSubAccounts: store.subAccountsConfig.get()?.enableAutoSubAccounts,
    }, AnalyticsEventImportance.high);
};
export const logInsufficientBalanceErrorHandlingStarted = ({ method, correlationId, }) => {
    logEvent('scw_sub_account.insufficient_balance.error_handling.started', {
        action: ActionType.unknown,
        componentType: ComponentType.unknown,
        method,
        correlationId,
        enableAutoSubAccounts: store.subAccountsConfig.get()?.enableAutoSubAccounts,
    }, AnalyticsEventImportance.high);
};
export const logInsufficientBalanceErrorHandlingCompleted = ({ method, correlationId, }) => {
    logEvent('scw_sub_account.insufficient_balance.error_handling.completed', {
        action: ActionType.unknown,
        componentType: ComponentType.unknown,
        method,
        correlationId,
        enableAutoSubAccounts: store.subAccountsConfig.get()?.enableAutoSubAccounts,
    }, AnalyticsEventImportance.high);
};
export const logInsufficientBalanceErrorHandlingError = ({ method, correlationId, errorMessage, }) => {
    logEvent('scw_sub_account.insufficient_balance.error_handling.error', {
        action: ActionType.error,
        componentType: ComponentType.unknown,
        method,
        correlationId,
        errorMessage,
        enableAutoSubAccounts: store.subAccountsConfig.get()?.enableAutoSubAccounts,
    }, AnalyticsEventImportance.high);
};
